### ISSUE TITLE
Search block: The block gets selected after adding a media item on another block

### ISSUE THREAD

**Issue by fluiddot** [WordPress/gutenberg#31351]
## Description

After adding the search block into a post, it gets automatically selected after adding or replacing media items on other media blocks like the image block.

## Step-by-step reproduction instructions

1. Add a search block
2. Add an image block
3. Select an image to be inserted in the image block
4. Observe that after the image is added the search block gets selected

## Expected behaviour

The image block should be kept selected after adding an image.

## Actual behaviour

The search block gets selected instead of the image block.

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/14905380/116595087-3c90bc80-a923-11eb-928b-a3bc3185a24f.mp4

## WordPress information
- WordPress version: N/A
- Gutenberg version: N/A
- Are all plugins except Gutenberg deactivated? N/A
- Are you using a default theme (e.g. Twenty Twenty-One)? N/A

## Device information
- Device: iPhone 8 (simulator)
- Operating system: iOS 14.2
- WordPress app version: `develop`

**Comment by fluiddot** [comment]
In some cases, I experienced a crash although I couldn't reproduce the exact steps. What I saw is that the scroll begins to jump from one place to another very quickly and then crashes, my impression was that different blocks were requesting focus and somehow the app enters in an infinite loop.

@AmandaRiu I ping you here as I saw that you recently enabled this block in this [PR](https://github.com/WordPress/gutenberg/pull/30783).
